web/packages/teleterm/src/ui/TopBar/NavigationMenu/NavigationMenu.tsx

```diff
@@ -18,75 +18,108 @@ import React, { useState, useRef } from 'react';
 import styled from 'styled-components';
 
 import Popover from 'design/Popover';
-import { MoreVert, OpenBox, Add } from 'design/Icon';
-import { Box, Text, Flex } from 'design';
+import { MoreVert, OpenBox, Add, Config } from 'design/Icon';
 
 import { useAppContext } from 'teleterm/ui/appContextProvider';
-import { DocumentsService } from 'teleterm/ui/services/workspacesService';
 import { TopBarButton } from 'teleterm/ui/TopBar/TopBarButton';
-import { RootClusterUri } from 'teleterm/ui/uri';
-
-import { useIdentity } from '../Identity/useIdentity';
+import { IAppContext } from 'teleterm/ui/types';
+import { Cluster } from 'teleterm/services/tshd/types';
 
 import { NavigationItem } from './NavigationItem';
 
-function getNavigationItems(
-  documentsService: DocumentsService,
-  clusterUri: RootClusterUri
-): {
-  title: string;
-  Icon: JSX.Element;
-  onNavigate: () => void;
-}[] {
+function useNavigationItems(): (
+  | {
+      title: string;
+      Icon: React.ComponentType<{ fontSize: number }>;
+      onNavigate: () => void;
+    }
+  | 'separator'
+)[] {
+  const ctx = useAppContext();
+  ctx.workspacesService.useState();
+  ctx.clustersService.useState();
+
+  const documentsService =
+    ctx.workspacesService.getActiveWorkspaceDocumentService();
+  const activeRootCluster = getActiveRootCluster(ctx);
+  const areAccessRequestsSupported =
+    !!activeRootCluster?.features?.advancedAccessWorkflows;
+
   return [
     {
-      title: 'New Access Request',
-      Icon: <Add fontSize={2} />,
-      onNavigate: () => {
-        const doc = documentsService.createAccessRequestDocument({
-          clusterUri,
-          state: 'creating',
-          title: 'New Access Request',
-        });
-        documentsService.add(doc);
-        documentsService.open(doc.uri);
+      title: 'Open Config File',
+      Icon: Config,
+      onNavigate: async () => {
+        const path = await ctx.mainProcessClient.openConfigFile();
+        ctx.notificationsService.notifyInfo(
+          `Opened the config file at ${path} in the default editor.`
```

Comment:
```suggestion
          `Opened the config file at ${path}.`
```

I didn't realize that the "in the default editor" part will almost always remain obscured by the path. In itself it doesn't add that much information after all.